content/ytt/docs/latest/how-it-works.md

```diff
@@ -0,0 +1,150 @@
+  ... 
+  ```
+  _(again, the details don't matter, yet; just know this is a "Template")_
+
+Now that we have a sense of each kind of input file, let's explore what happens at each step in the pipeline.
+
+### Step 1: Calculate Data Values
+
+First,
+
+1. from the input files, grab all the "Data Values" templates;
+1. evaluate those templates, yielding a list of documents;
+1. merge those documents, [in order](ytt-data-values.md#splitting-data-values-into-multiple-files). That is, start with the first document and then overlay the second one onto it; then overlay the third document on top of _that_, and so on...
+
+The result of all this is the final set of values that will be available to other templates: the "final Data Values", [above](#the-pipeline).
+
+### Step 2: Evaluate Templates
+
+Next, evaluate the remaining templates:
+1.  "evaluate" means executing all of the Starlark code: loops are run, conditionals decided, expressions evaluated.
+1.  one notable exception are the overlay annotations (i.e. those that start with `@overlay/...`), these are deferred until the next step.
+1.  a template accesses input variables (i.e. the Data Values calculated in the previous step) via the [`@ytt:data` module](lang-ref-ytt.md#data);
+    
+The result of all this evaluation is a set of YAML documents, customized by the inputs (shown as "Evaluated DocSet" in the diagram, [above](#the-pipeline)).
+
+### Step 3: Apply Overlays
+
+Next...
+
+1. split the "Evaluated Document Set" into two groups:
+    - "Overlay Documents" — documents annotated with [`@overlay/match...`](lang-ref-ytt-overlay.md#overlays-as-files)
+    - "Template Documents" — the remaining documents (remember, these are what will ultimately be the output of this pipeline)
+1. apply each "Overlay Document" on top of the set of "Template Documents".\
+   You can think of each overlay as like a SQL `update` command:
+   - the value of it's `by` argument is like a `where` clause that selects over the whole collection of "Template Documents". For example, 
+     ```yaml
+     #@overlay/match by=overlay.subset({"kind": "Deployment"}), ...
+     ---
+     ```
+      selects all of the documents which contain a key `"kind"` whose value is `"Deployment"`
+   - for each of the documents selected, apply the overlay on top of it. This is like a series of `set` clauses, each updating a portion of the document. For example,
+     ```yaml
+     #@overlay/match by=overlay.subset({"kind": "Deployment"}), ...
+     ---
+     #@overlay/match-child-defaults missing_ok=True
+     metadata:
+       labels:
+         app: frontend
+     ```
+     sets each "Deployment"'s `metadata.labels.app` to be `"frontend"`.
+1. repeat that process for each "Overlay Document", [in order](lang-ref-ytt-overlay.md#overlay-order).
+   
+The result is (shown as "Output DocSet" in the diagram, [above](#the-pipeline)) — the finalized set of YAML documents, in memory. Which leaves one last step...
+
+### Step 4: Print
```

Comment:
instead of calling this print, may be it should be "serializing". one thing we should probably mention is that because we are dealing with structures, it's always going to come out as legitimate yaml.

Reply:
Yeah, I was trying out the verb from the codebase. If the intro can support the would be contributor — without damaging _this_ explanation — it's all the more valuable.

But re-reading it with your observation, I see that "serializing" is going to be more familiar in this context and convey more important things (i.e. the structure will be preserved).